packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputModelType.cs

```diff
@@ -79,12 +80,15 @@ public IReadOnlyDictionary<string, InputModelType> DiscriminatedSubtypes
             get => _discriminatedSubtypes ??= new Dictionary<string, InputModelType>();
             internal set
             {
-                if (value is null || value.Count == 0)
+                if (value is null || DiscriminatorProperty == null)
                     return;
 
                 _discriminatedSubtypes = new Dictionary<string, InputModelType>(value);
-                var cleanBaseName = Name.ToCleanName();
-                _discriminatedSubtypes.Add("unknown",
+
+                if (DiscriminatorValue != UnknownDiscriminatorValue)
```

Comment:
Do we need to support adding an "Unknown" discriminator in the tsp? This has a semantically special meaning to our generator where it is the fallback, but if it is defined in the tsp, it seems like they would want to match on the literal string "Unknown". I think we can probably leave this off for now until we here of a use case for it.

Reply:
Yes this makes sense and I agree that it is a valid scenario even if it's unlikely.